markup/goldmark/convert.go

```diff
@@ -262,6 +276,7 @@ func (c *goldmarkConverter) Parse(ctx converter.RenderContext) (converter.Result
 		reader,
 		parser.WithContext(pctx),
 	)
+	doc.OwnerDocument().AddMeta("documentID", c.ctx.DocumentID)
```

Comment:
I have scabbed the code, and it looks like this is the first time we use the value of `DocumentID`.

Currently, this value is derived from the content's backing file (`f.UniqueID()`), blank if not. For now, let's skip the discussion about that method, but this means that 

1. Pages without a content file will not have an ID.
2. Pages created via content adapters will have the same file ID (I assume)

I suggest that we where the `converter.DocumentContext` is created creates a new ID which is the XxHashed version of `page.Path`.